httpMiddleware.go

```diff
@@ -0,0 +1,199 @@
+package main
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"strconv"
+	"time"
+
+	"github.com/palantir/go-githubapp/githubapp"
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/rs/zerolog"
+)
+
+type statusCodeResponseWriter struct {
+	http.ResponseWriter
+	statusCode int
+}
+
+func (w *statusCodeResponseWriter) WriteHeader(code int) {
+	w.statusCode = code
+	w.ResponseWriter.WriteHeader(code)
+}
+func InitializeResponder(ctx context.Context) context.Context {
+	var responder func(http.ResponseWriter, *http.Request)
+	return context.WithValue(ctx, responderKey{}, &responder)
+}
+
+type responderKey struct{}
+type LoggerFunc func(msg string, m map[string]interface{})
+
+func addContextMiddleware(logger zerolog.Logger, h http.Handler) http.Handler {
+	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		//eventType := r.Header.Get("X-GitHub-Event")
+		//deliveryID := r.Header.Get("X-GitHub-Delivery")
+
+		ctx := logger.With()
+		//ctx = ctx.Str("github_event_type", eventType)
+		//ctx = ctx.Str("github_delivery_id", deliveryID)
+
+		l := ctx.Logger()
+
+		r = r.WithContext(l.WithContext(r.Context()))
+
+		h.ServeHTTP(w, r)
+	})
+}
+
+func loggerMiddleware(logger LoggerFunc, h http.Handler) http.Handler {
+	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		statusWriter := &statusCodeResponseWriter{w, http.StatusOK}
+		start := time.Now()
+		h.ServeHTTP(statusWriter, r)
+
+		// request duration in miliseconds
+		duration := time.Since(start).Nanoseconds() / 1e6
```

Comment:
We can use [`time.Duration.Milliseconds`](https://godoc.org/time#Duration.Milliseconds) to do this instead.
It is a newer addition (Go 1.14 I think) of the standard lib, so I'm guessing this is from a snippet of earlier origin :) 

```suggestion
		duration := time.Since(start).Milliseconds()
```